crates/matrix-sdk/src/sliding_sync/mod.rs

```diff
@@ -483,21 +483,33 @@ impl SlidingSync {
             if self.inner.sticky.read().unwrap().data().extensions.e2ee.enabled == Some(true) {
                 debug!("Sliding Sync is sending the request along with outgoing E2EE requests");
 
-                let (e2ee_uploads, response) =
-                    futures_util::future::join(self.inner.client.send_outgoing_requests(), request)
-                        .await;
-
-                if let Err(error) = e2ee_uploads {
+                // If the sliding sync request fails faster than the e2ee requests get their
+                // responses, we want to fail as early as possible, so as to invalidate the
+                // sliding sync session as early as possible.
+                //
+                // Also, we don't want an interruption in the sliding sync request to abort
+                // processing of the e2ee response.
+                //
+                // For those reasons, start the e2ee requests in a background task.
+
+                let client = self.inner.client.clone();
+                let e2ee_uploads = spawn(async move { client.send_outgoing_requests().await });
+
+                // Wait on the sliding sync request success or failure early.
+                let response = request.await?;
+
+                // Then only log the e2ee response, if needs be.
+                if let Err(error) = e2ee_uploads.await {
```

Comment:
Hmm, this means that if the SS requests fails we won't ever hit this log line? Perhaps we should move this into the spawn? It also means that we're still going to wait for the e2ee requests to be done before we continue with the next SS request. I guess that's fine considering that the loops will be split.